src/main/kotlin/parts/wisdom/deansher/arcraider/Arc.kt

```diff
@@ -0,0 +1,124 @@
+package parts.wisdom.deansher.arcraider
+
+import parts.wisdom.arcraider.ArcColor
+import parts.wisdom.arcraider.SerializedGrid
+import parts.wisdom.arcraider.ArcPair as SerializedPair
+import parts.wisdom.arcraider.Task as SerializedTask
+
+/**
+ * A grid with dimensions `width, height`, indexed by `[x, y]` where `x in 0 until width` goes left to right
+ * and `y in 0 until height` goes top to bottom.
+ */
+data class ArcGrid(
+    val width: Int,
+    val height: Int,
+    /** Stores `(x, y)` at `y * width + x` */
+    private val data: Array<ArcColor>
```

Comment:
oh this is clever, nice way to have it indexable and iterable as a single list. Currently this is private though? How does one just iterate this list?